src/pool/srv_iv.c

```diff
@@ -42,16 +42,18 @@ static int
 pool_iv_value_alloc_internal(d_sg_list_t *sgl)
 {
 	uint32_t	buf_size;
-	uint32_t	pool_nr;
+	uint32_t	nnodes, ndomains, ntgts;
 	int		rc;
 
 	rc = daos_sgl_init(sgl, 1);
 	if (rc)
 		return rc;
 
-	/* XXX Let's use primary group  + 1 domain per target now. */
-	crt_group_size(NULL, &pool_nr);
-	buf_size = pool_iv_ent_size((int)pool_nr * 2 * 10);
+	crt_group_size(NULL, &nnodes);
+	/* currently with 1 domain per node, see init_pool_metadata */
+	ndomains = nnodes;
+	ntgts = nnodes * dss_tgt_nr;
+	buf_size = pool_iv_ent_size(ndomains + nnodes + ntgts);
```

Comment:
Thanks Xuezhao. That did the trick for me and we should land it IMO.
This still need to be discussed with LiWei since i am not sure how it will interfere with the self bootstrapping and dynamic group size.

Reply:
Ah, yes, when group size increase it still with problem. Need some other refining for that later...

Reply:
That's true. Need more thoughts.